src/main/java/org/opensearch/flowframework/workflow/AbstractUpdatePipelineStep.java

```diff
@@ -0,0 +1,132 @@
+        Set<String> optionalKeys = Set.of(MODEL_ID);
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs,
+                params
+            );
+
+            String pipelineId = (String) inputs.get(PIPELINE_ID);
+            String configurations = (String) inputs.get(CONFIGURATIONS);
+
+            // Special case for processors that have arrays that need to have the quotes around or
+            // backslashes around strings in array removed
+            String transformedJsonStringForStringArray = ParseUtils.removingBackslashesAndQuotesInArrayInJsonString(configurations);
+
+            byte[] byteArr = transformedJsonStringForStringArray.getBytes(StandardCharsets.UTF_8);
+            BytesReference configurationsBytes = new BytesArray(byteArr);
+
+            String pipelineToBeCreated = this.getName();
+            ActionListener<AcknowledgedResponse> putPipelineActionListener = new ActionListener<>() {
+
+                @Override
+                public void onResponse(AcknowledgedResponse acknowledgedResponse) {
+
+                    // Not necessary to update state index entry since the resource ID remains unchaged
+                    String resourceName = getResourceByWorkflowStep(getName());
+                    logger.info("Successfully updated resource: {}", pipelineId);
+                    createPipelineFuture.onResponse(
+                        new WorkflowData(Map.of(resourceName, pipelineId), currentNodeInputs.getWorkflowId(), currentNodeInputs.getNodeId())
+                    );
+                }
+
+                @Override
+                public void onFailure(Exception ex) {
+                    Exception e = getSafeException(ex);
+                    String errorMessage = (e == null ? "Failed step " + pipelineToBeCreated : e.getMessage());
+                    logger.error(errorMessage, e);
+                    createPipelineFuture.onFailure(new WorkflowStepException(errorMessage, ExceptionsHelper.status(e)));
+                }
+
+            };
+
+            if (pipelineToBeCreated.equals(UpdateSearchPipelineStep.NAME)) {
+                PutSearchPipelineRequest putSearchPipelineRequest = new PutSearchPipelineRequest(
+                    pipelineId,
+                    configurationsBytes,
+                    XContentType.JSON
+                );
+                clusterAdminClient.putSearchPipeline(putSearchPipelineRequest, putPipelineActionListener);
+            } else {
+                PutPipelineRequest putPipelineRequest = new PutPipelineRequest(pipelineId, configurationsBytes, XContentType.JSON);
+                clusterAdminClient.putPipeline(putPipelineRequest, putPipelineActionListener);
+            }
```

Comment:
I really don't like the superclass needing to know which subclass to call here.  OK for this PR but expect me to refactor this later :)

Reply:
Ill add an abstract method to execute the put pipeline request and have the subclasses implement that. This should remove the need to check for what type of step is executing this